app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -50,10 +47,14 @@ def plan(smart_proxy, options = {})
 
           unless repository_ids.blank?
             sequence do
-              need_updates.each do |repo|
-                plan_action(Pulp::Repository::Refresh, repo, capsule_id: smart_proxy_service.smart_proxy.id)
-              end
-              plan_action(ConfigureCapsule, smart_proxy_service.smart_proxy, environment, content_view, repository)
+              options = {}
+              options[:smart_proxy_id] = smart_proxy.id
+              options[:environment_id] = environment_id if environment_id
+              options[:content_view_id] = content_view_id if content_view_id
+              options[:repository_id] = repository_id if repository_id
+
+              plan_action(Pulp::Repository::RefreshNeeded, options)
+              plan_action(CreateRepos, smart_proxy, environment, content_view, repository)
```

Comment:
I think refresh needs to come after createRepos, otherwise you may get 404s on the refresh